### PR TITLE
Handle oversized PES packet lengths

### PR DESCRIPTION
**Pull request by Ponsaille**
Closes #15

## Summary

This PR fixes `PES_packet_length` serialization for PES packets whose header + payload length exceeds `0xFFFF`.

Previously, the muxer always wrote the computed PES packet length into a 16-bit field. For large video PES packets, this could overflow/truncate the encoded length and break playback. Per MPEG-TS semantics, `PES_packet_length = 0` represents an unspecified length and is allowed for video PES packets in transport streams.

Following the discussion in #15, this PR uses that unspecified length form for oversized only for video PES packets.

It also adds an error when there is an oversized audio PES packet, I know we don't expect this to happen but I guess it is a nice hint for someone doing something crazy in the future?

## Changes

- Pass `track_type` from `Engine.put_frame/5` into `PES.serialize/5`
- Keep the normal encoded `PES_packet_length` when it fits in 16 bits
- Encode oversized video PES packets with `PES_packet_length = 0`
- Raise for oversized non-video PES packets, since unspecified PES length is only valid for video streams
- Add focused tests for oversized video, oversized audio, and regular audio PES packet lengths

## Validation

- `mix test test/muxer/pes_test.exs`

### CODE REVIEW

File: test/muxer/pes_test.exs [hunk 1]
```diff
@@ -5,6 +5,32 @@ defmodule Membrane.MPEGTS.Muxer.PESTest do
 
   alias Membrane.MPEGTS.Muxer.PES
 
+  test "oversized video PES packets use zero packet length" do
+    payload = :binary.copy(<<0>>, 0xFFFF)
```
Comment (varsill) [inline test/muxer/pes_test.exs:9]: [NIT] I would make it slightly larger than the boundary value of `0xFFFF` as at the first sight it might look slightly weird that we are putting a payload of length `0xFFFF` (which is `<= @max_pes_packet_length`), but we end up with `pes_packet_length == 0`
(ofcourse the reason for that is that we are always appending ES header to the payload and we are comparing the total length of the payload and the header against `@max_pes_packet_length`)

**Review by varsill — APPROVED** [review]
Thank you very much for the contribution once again! I will implement the suggested change and release new version of the plugin in a moment.
